alphastats/llm/uniprot_utils.py

```diff
@@ -215,43 +210,69 @@ def extract_data(data: Dict) -> Dict:
     ]
     extracted["tissueSpecificity"] = tissue_specificities
 
-    # 11. Protein Features
-    features = [
-        {
-            "type": feature["type"],
-            "description": feature["description"],
-            "location_start": feature["location"]["start"]["value"],
-            "location_end": feature["location"]["end"]["value"],
-        }
-        for feature in data.get("features", [])
-    ]
-    extracted["features"] = features
-
-    # 12. References
-    references = [
-        {
-            "authors": ref["citation"].get("authors", []),
-            "title": ref["citation"].get("title", ""),
-            "journal": ref["citation"].get("journal", ""),
-            "publicationDate": ref["citation"].get("publicationDate", ""),
-            "comments": [c["value"] for c in ref.get("referenceComments", [])],
-        }
-        for ref in data.get("references", [])
-    ]
-    extracted["references"] = references
+    ## 11. Protein Features
+    # features = [
+    #    {
+    #        "type": feature["type"],
+    #        "description": feature["description"],
+    #        "location_start": feature["location"]["start"]["value"],
+    #        "location_end": feature["location"]["end"]["value"],
+    #    }
+    #    for feature in data.get("features", [])
+    # ]
+    # extracted["features"] = features
+
+    ## 12. References
+    # references = [
+    #    {
+    #        "authors": ref["citation"].get("authors", []),
+    #        "title": ref["citation"].get("title", ""),
+    #        "journal": ref["citation"].get("journal", ""),
+    #        "publicationDate": ref["citation"].get("publicationDate", ""),
+    #        "comments": [c["value"] for c in ref.get("referenceComments", [])],
+    #    }
+    #    for ref in data.get("references", [])
+    # ]
+    # extracted["references"] = references
 
     # 13. Cross References
-    cross_references = [
+    # cross_references = [
+    #    {
+    #        "database": ref["database"],
+    #        "id": ref["id"],
+    #        "properties": {
+    #            prop["key"]: prop["value"] for prop in ref.get("properties", [])
+    #        },
+    #    }
+    #    for ref in data.get("uniProtKBCrossReferences", [])
+    #    if ref['database'] not in ['GO', 'Reactome']
+    # ]
+    # extracted["crossReferences"] = cross_references
+
+    # 14. Pathway references
+    pathway_references = [
         {
             "database": ref["database"],
             "id": ref["id"],
-            "properties": {
-                prop["key"]: prop["value"] for prop in ref.get("properties", [])
-            },
+            "pathway": ref.get("properties")[0]["value"],
+        }
+        if ref["database"] == "Reactome"
+        else {
+            "database": ref["database"]
+            + " "
+            + {"P": "Pathway", "C": "Component", "F": "Function"}[
```

Comment:
disclaimer: no clue what goes on here in terms of biology :-D

Reply:
GO has three categories of terms and uniprot encodes them by appending a single letter code to the annotation name. I split them here and in the next PR also store them separately so they become individually selectable.